amm/contracts/router_v2/stable_pool.rs

```diff
@@ -129,7 +124,7 @@ impl StablePool {
                 self.transfer_tokens_back(&amounts, to, wnative_idx)?;
                 res
             }
-            None => {
+            Err(_) => {
```

Comment:
To be 100% clean it should be done only when matching a specific variant and propagate error upwards if some other error is present (same for the second method).

From what I see it's not possible to get any other error than `InvalidToken`, so your call

Reply:
NIT: Maybe even cleaner would be to change `fn wnativre_idx(&self, wnative: AccountId)` to return `Option<usize>`? After this refactor there's only one place where we want to return `InvalidToken` error when wnative is not present.

Reply:
I like @JanKuczma's suggestion - if `wnative_idx` returns an error, it always has to be `InvalidToken`, and inside the fn we can see that it converts the `Option` to `Result` as the last operation. So let's just return an `Option` and wherever `wnative_idx` was used to potentially throw an error, just map the `Option` to `Result` there.

Reply:
Yes, that's even better